src/main/java/com/nirima/jenkins/plugins/docker/DockerContainerWatchdog.java

```diff
@@ -440,4 +465,12 @@ private void cleanUpSuperfluousComputer(Map<String, Node> nodeMap, ContainerNode
         }
     }
 
+    private void checkForTimeout(Instant startedTimestamp) {
+        final Instant now = this.clock.instant();
+        Duration runtime = Duration.between(startedTimestamp, now);
+        
+        if (runtime.compareTo(PROCESSING_TIMEOUT_IN_MS) > 0) {
+            throw new WatchdogProcessingTimeout();
```

Comment:
I would guess that, in order to provide the information I requested [above](https://github.com/jenkinsci/docker-plugin/pull/658/commits/3320a9a816cb8d1cf3f2003e46cea8da33a67b51#diff-8933db77131fb3a413685c224649363aR154) it may be useful to store `startedTimestamp`, `now` and `PROCESSING_TIMEOUT_IN_MS` in the exception so that this information can be retrieved later for logging.
(I'm sure there's other ways of achieving the same effect; this is just one possibility)